### DISCUSSION TITLE
Time format: precision not clarified in the docs

### DISCUSSION THREAD

**Discussion by jamescridland** [skymethod/op3#10] [General]
The API asks for the date/time as:

```
string <ISO 8601 or relative duration>
Example: start=2022-09-15T14:00:52.709Z
```

ISO8601, and [RFC3222](https://datatracker.ietf.org/doc/html/rfc3339#section-5), doesn't specify the precision in microseconds; but, after experimentation, it appears OP3 requires a seconds precision of exactly three decimal points, no more, no less.

`https://op3.dev/api/1/redirect-logs?start=2022-11-20T00:00:00.000Z&format=json&token=preview07ce` works

`https://op3.dev/api/1/redirect-logs?start=2022-11-20T00:00:00.000000Z&format=json&token=preview07ce` (PHP's default) doesn't work

`https://op3.dev/api/1/redirect-logs?start=2022-11-20T00:00:00Z&format=json&token=preview07ce` (no microseconds) doesn't work.

Would suggest the docs read...

```
string <ISO 8601 to a seconds precision of three decimal points, or relative duration>
```

... or something similar.

If it helps...

```php
$datefrom="2022-11-20T22:12:13.2";
$date=substr(date_format(date_create($datefrom), "Y-m-d\TH:i:s.u"),0,23).'Z';
```

...will output a normalised date/time format that works for this API.

**Comment by johnspurlock-skymethod** [comment]
We can probably be more liberal parsing these user inputs here, in order to accommodate other programming language defaults - just pushed a change 052cb760d4e080591ccbe98db094550a4dbfe0c7 to production, now the `start` and `end` params accept iso8601 date strings with any amount of decimals (or none at all)

**Comment by jamescridland** [comment]
Thanks! :)
